extensions/cache/deployment/src/main/java/io/quarkus/cache/deployment/CacheAnnotationsTransformer.java

```diff
@@ -0,0 +1,46 @@
+package io.quarkus.cache.deployment;
+
+import static io.quarkus.cache.deployment.CacheDeploymentConstants.AUGMENTED_CACHE_INVALIDATE;
+import static io.quarkus.cache.deployment.CacheDeploymentConstants.AUGMENTED_CACHE_INVALIDATE_ALL;
+import static io.quarkus.cache.deployment.CacheDeploymentConstants.AUGMENTED_CACHE_LOAD;
+import static io.quarkus.cache.deployment.CacheDeploymentConstants.AUGMENTED_CACHE_STORE;
+import static io.quarkus.cache.deployment.CacheDeploymentConstants.CACHE_INVALIDATE;
+import static io.quarkus.cache.deployment.CacheDeploymentConstants.CACHE_INVALIDATE_ALL;
+import static io.quarkus.cache.deployment.CacheDeploymentConstants.CACHE_LOAD;
+import static io.quarkus.cache.deployment.CacheDeploymentConstants.CACHE_NAME_PARAMETER_NAME;
+import static io.quarkus.cache.deployment.CacheDeploymentConstants.CACHE_STORE;
+
+import org.jboss.jandex.AnnotationTarget.Kind;
+import org.jboss.jandex.AnnotationValue;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.MethodInfo;
+
+import io.quarkus.arc.processor.AnnotationsTransformer;
+
+/**
+ * @author Gwenneg Lepage
+ */
+public class CacheAnnotationsTransformer implements AnnotationsTransformer {
```

Comment:
This class exists because the `cacheName` parameter from a user-added `@CacheLoad` annotation (for example) can be valued or empty. The real cache name is determined at build time (see `CacheNameResolver`) and stored as a parameter of an `@AugmentedCacheLoad` annotation (still the same example).